soc: mcxw: Fix boot and enable timers

decsny opened the pull request:
- Enable some of the timers on MCXW71 (TPM, LPTMR, WDOG : PWM, counter, watchdog)

- Fix boot issues caused by changes to upstream platform init while waiting for merge of #77434

- Some other minor adjustments

Commit:
drivers: pwm_mcux_tpm: support pause level feature

Support TPM devices with the pause level selection feature

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
soc: nxp: mcxw: Switch to soc_reset_hook

Base enablement of MCXW merged while z_arm_platform_init was being
deprecated, resulting situation is now that no platform init is
happening, fix by converting to use soc_reset_hook.

Also fix a comment that said the core was being set to 40 MHz, when it
is actually being set up to 96 MHz.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
dts: nxp: mcxw71: Add peripheral bridge definition

Add peripheral bridge definitions to DT, this also fixes the base
address of the GPIO peripherals which were faulting in the driver due to
the wrong reg address.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
boards: frdm_mcxw71: Rename LED to blue

The LED is blue, not green.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
soc: mcxw71: Add TPM support

Add DT node and clocking of TPM peripherals, which are used for PWM.

Also change the soc clock enable code to not use preprocessor
conditionals.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
boards: frdm_mcxw71: Enable TPM and pwm_leds

Enable TPM0 on FRDM_MCXW71 which is connected to an RGB LED.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
dts: nxp: mcxw71: Add WDOG nodes to DT

Add watchdog nodes to MCXW71 SOC DT, and enable wdog0 alias.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
dts: nxp: mcxw71: Use genetic unit names

Use generic unit names as recommended by the DT spec.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
dts: nxp: mcxw71: Add LPTMR0 and LPTMR1

Add nodes for LPTMR. This is sufficient to enable their use as counter
devices when set to status okay.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
boards: frdm_mcxw71: Enable LPTMR

Enable LPTMR as counter, enable lptmr0 in board DTS and enable lptmr1 in
basic counter api test.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

danieldegrasse reviewed (commented):
YAML and doc pages for this board need to be updated to indicate support for counter and watchdog. Otherwise, PR LGTM

Commit:
boards: frdm_mcxw71: Add timers to doc & yaml

Add timers that have been enabled to board doc table

Add timers that have been enabled to board twister yaml

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

axelnxp commented on soc/nxp/mcx/mcxw/mcxw71_platform_init.S:
```diff
@@ -16,13 +16,11 @@
-SECTION_SUBSEC_FUNC(TEXT,_reset_section,z_arm_platform_init)
-
-.z_arm_platform_init:
+.soc_reset_hook:
```
I would suggest to have a dedicated function for the ECC ram init, and have `soc_reset_hook` calling it.
The reason behind is we may need to call the ECC ram init as a standalone procedure, outside the early init.
For instance, at wake up from power down mode, if an ECC ram bank has been shutdown, we need to go through this procedure again, so we could call it directly.

This is why in #79060, we implemented `soc_reset_hook` in C, and the ECC init function separately.
Having `soc_reset_hook` in C would likely be "easier" to maintain and extend if needed.

axelnxp replied:
Where is placed the stack ? STCM3 and STCM4 are non-ecc (range 0x30010000 - 0x30019FFF)

decsny replied:
and also, to answer your question, the stack can be anywhere, it just goes at the end of the image. Whether or not it goes in non-ecc range is just depending on the size of the image.
